commands/peers/addpeer.go

```diff
@@ -35,6 +35,20 @@ func addPeerHandler(w http.ResponseWriter, r *http.Request) {
 		req.Name = req.Addresses[0]
 	}
 
+	local_node := false
+	for _, addr := range req.Addresses {
+		local, err := utils.IsLocalAddress(addr)
+		if err != nil || local == true {
```

Comment:
This was deliberate. This err can be returned by call to `net.LookupIP` (internally done in `IsLocalAddress`) which can fail on any one of the multiple addresses which is OK because the attach peer API allows specifying multiple addresses. The goal here is to identify if the peer is local node. If DNS resolution fails one one of the addresses, that should be okay at this point in time. If the host is not reachable, it'll fail later during the RPC call right. What do you think ?